README.md

```diff
@@ -193,3 +193,61 @@ $ telnet 192.168.99.100 1234
 
 This `-p` argument links port 4321 on the container to port 1234 on the VM network adapter. It must appear after `docker run` and before `dmscid/plankton`. This allows us to connect to the container from outside of the VM, in this case using Telnet. The `192.168.99.100` IP is the IP of the VM on the bridge network between the host and the VM. VirtualBox will typically use this IP when available, but it may be different on your system.
 
+## Remote access to devices
+
+*Please note that this functionality should only be used on a trusted network.*
+
+Besides the device specific protocols, the device can be made accessible from the outside via JSON-RPC over ZMQ. This can be achieved by passing the `-r` option with a `host:port` string to the simulation:
+
+```
+$ python simulation.py -r 127.0.0.1:10000 -d chopper -- -p SIM:
+```
+
+Now the device can be controlled via the `control.py`-script in a different terminal window. The service can be queried to show the available objects using the `-l` or `--list-objects` flag:
+
+```
+$ python control.py --ip 127.0.0.1 --port 10000 --list-objects
+```
+
+The `ip` and `port` parameters default to the values shown here, so they will be omitted in the following examples. To get information on the API of an object, the `-a` or `--show-api` option can be used in conjunction with a device name obtained from the previous command:
+
+```
+$ python control.py -a device
+```
+
+This will output a list of properties and methods which is available for remote access. This may not comprise the full interface of the object depending on the server side configuration. Obtaining the value of a property is done like this:
+
+```
+$ python control.py device state
+```
+
+The same syntax is used to call methods without parameters:
+
+```
+$ python control.py device initialize
+```
+
+To set a property to a new value, the value has to be supplied on the command line:
+
+```
+$ python control.py device targetSpeed 100
+$ python control.py device start
+```
+
+Only numeric types and strings can be used as arguments via the `control.py`-script. The script always tries to convert parameters to `int` first, then to `float` and leaves it as `str` if both fail. For other types and more control over types, it's advised to write a Python script instead using the tools provided in `core.rpc_client` which makes it possible to use the remote objects more or less transparently. An example to control the chopper:
+
+```
+from time import sleep
+from core.rpc_client import ZMQJSONRPCConnection, get_remote_object_collection
```

Comment:
I don't think I'm a fan of the class names.

As far as I can tell, from the user's perspective, ZMQ and JSONRPC are implementation details that are never seen or interacted with... except for that a lot of things have it in the name. Additionally, since every module is a namespace in Python, it seems a bit redundant to have such long prefixes. And `ExposedObject` and `ExposedObjectCollection` don't follow that format anyway (I rather like the way those are named).

How would you feel about renaming things to, for example:
```
JSONRPCServerSideException  ->  RemoteException
JSONRPCProtocolException    ->  ProtocolException
ZMQJSONRPCConnection        ->  ControlClient
JSONRPCObjectProxy          ->  ObjectProxy
ZMQJSONRPCServer            ->  ControlServer
```
And maybe also 
```
rpc_client.py    ->    control_client.py
rpc_server.py    ->    control_server.py
```

Additionally, I would make `get_remote_object_collection` and `get_remote_object` members of `ZMQJSONRPCConnection` (or rather `ControlClient`). 

That way, user facing code becomes:

```python
from plankton.core.control_client import ControlClient

remote = ControlClient(host='...', port='...')
chopper = remote.get_object('chopper')

# or maybe even just...
chopper = remote['chopper']
```

This feels a lot more intuitive / readable to me.